src/AzureStack/Subscriptions/Subscriptions.Tests/src/SubscriptionTests.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+//
+
+using Microsoft.AzureStack.Management.Subscriptions;
+using Microsoft.AzureStack.Management.Subscriptions.Models;
+using System;
+using Xunit;
+
+namespace Subscriptions.Tests
+{
+    public class SubscriptionTests : SubscriptionsTestBase
+    {
+
+        private void ValidateSubscription(Subscription subscription) {
+            // Subscription
+            Assert.NotNull(subscription);
+            Assert.NotNull(subscription.Id);
+            Assert.NotNull(subscription.DisplayName);
+            Assert.NotNull(subscription.SubscriptionId);
+            Assert.NotNull(subscription.TenantId);
+            Assert.NotNull(subscription.State);
+            Assert.NotNull(subscription.OfferId);
+        }
+
+        [Fact]
+        public void TestListSubscriptions()
+        {
+            RunTest((client) =>
+            {
+                var subscriptions = client.Subscriptions.List();
+
+                subscriptions.ForEach(ValidateSubscription);
+            });
+        }
+
+        [Fact]
+        public void TestGetSubscription()
+        {
+            RunTest((client) =>
+            {
+                var subscriptions = client.Subscriptions.List();
+                subscriptions.ForEach(subscription =>
+                {
+                    ValidateSubscription(subscription);
+                    client.Subscriptions.Get(subscription.SubscriptionId);
+                });
+            });
+        }
+
+
+        //[Fact]
```

Comment:
@bganapa any reason you want to keep commented code?

Reply:
I have added skip = true, 
        //The test would fail with the following as we try to create the subscription as ServicePrincipal Identity which is not allowed
        //The request identity does not contain valid name claim type and is not allowed to create new subscription.
//The call has been validated with user identity manually